core/shared/src/main/scala/scalaz/zio/ZManaged.scala

```diff
@@ -292,6 +292,33 @@ final case class ZManaged[-R, +E, +A](reserve: ZIO[R, E, Reservation[R, E, A]])
       }
     }
 
+  /**
+   * Creates a `ZManaged` value that acquires the original resource in a fiber,
+   * and provides that fiber. The finalizer for this value will interrupt the fiber
+   * and run the original finalizer.
+   */
+  final def fork: ZManaged[R, Nothing, Fiber[E, A]] =
+    ZManaged {
+      for {
+        finalizer <- Ref.make[ZIO[R, Nothing, _]](UIO.unit)
+        // The reservation phase of the new `ZManaged` runs uninterruptibly;
+        // so to make sure the acquire phase of the original `ZManaged` runs
+        // interruptibly, we need to create an interruptible hole in the region.
+        fiber <- ZIO.interruptibleMask { restore =>
```

Comment:
@jdegoes would love another look at this operator for validation. I used `interruptibleMask` to allow the caller to run the reserve step interruptibly rather than `.interruptible`

Reply:
This is clever and looks correct to me. 

One thing to keep in mind is that this fiber will inherit the interruptibility settings of its parent, at the moment when it is forked.

Since this is done in `reserve`, I believe we can therefore simplify the code to just use `interruptible` rather than `interruptibleMask`, because we know what the parent interruptibility status will be (right?).

Reply:
So we can assume that 99% of the cases would use an uninterruptible `reserve`, but, that's not always true, as `reserve` can also be run manually for advanced manipulations. Which is why I thought `interruptibleMask` is more compositional.

Reply:
Ah, ok. Works for me. 👍